libraries/AC_AttitudeControl/AC_PosControl.cpp

```diff
@@ -815,24 +814,24 @@ void AC_PosControl::input_pos_vel_accel_z(Vector3f& pos, Vector3f& vel, const Ve
     }
 
     // adjust desired alt if motors have not hit their limits
-    update_pos_vel_accel_z(_pos_target, _vel_desired, _accel_desired, _dt, _limit_vector);
+    update_pos_vel_accel(_pos_target.z, _vel_desired.z, _accel_desired.z, _dt, _limit_vector.z);
 
-    shape_pos_vel_accel_z(pos, vel, accel,
-        _pos_target, _vel_desired, _accel_desired,
-        0.0f, _vel_max_down_cms, _vel_max_up_cms,
-        -constrain_float(accel_z_cmss, 0.0f, 750.0f), accel_z_cmss,
-        _tc_z_s, _dt);
+    shape_pos_vel_accel(pos, vel, accel,
+                        _pos_target.z, _vel_desired.z, _accel_desired.z,
+                        0.0f, _vel_max_down_cms, _vel_max_up_cms,
+                        -constrain_float(accel_z_cmss, 0.0f, 750.0f), accel_z_cmss,
+                        _tc_z_s, _dt);
 
-    update_pos_vel_accel_z(pos, vel, accel, _dt, Vector3f());
+    update_pos_vel_accel(pos, vel, accel, _dt, 0);
 }
 
 /// set_alt_target_with_slew - adjusts target up or down using a commanded altitude in cm
 ///     using the default position control kinimatic path.
 void AC_PosControl::set_alt_target_with_slew(const float& pos)
 {
-    Vector3f pos_3f = Vector3f{0.0f, 0.0f, pos};
-    Vector3f zero;
-    input_pos_vel_accel_z(pos_3f, zero, zero);
+    float posf = pos;
+    float zero = 0;
```

Comment:
@lthall this one too

Reply:
Again, this is a stationary position target with zero velocity and acceleration. Therefore we do not need to update the final target.